cpp/tests/utilities/column_utilities.cu

```diff
@@ -42,7 +42,7 @@ void expect_column_properties_equal(cudf::column_view lhs, cudf::column_view rhs
      EXPECT_EQ(lhs.nullable(), rhs.nullable());
   }
   EXPECT_EQ(lhs.has_nulls(), rhs.has_nulls());
-  EXPECT_EQ(lhs.num_children(), rhs.num_children());
+  //EXPECT_EQ(lhs.num_children(), rhs.num_children());
```

Comment:
Need to resolve this issue and remove commented out code.

Reply:
Yep, this was just a temporary workaround to make the tests pass. The issue is that strings columns created by empty_like do not compare equal to those created by make_empty_strings_column!

Reply:
I saw that.  Can you fix that discrepancy in this PR?

Reply:
Sure, but I'm not sure what the correct behavior should be:
- empty_like should NOT copy child columns for strings
- make_empty_strings_column SHOULD allocate child columns
- both are fine and expect_column_properties_equal is too strict
I was hoping for a comment from @jrhemstad and @davidwendt on what the correct behavior should be

Reply:
Initially, I had set up an empty strings column to have empty children. But I settled on not making the child columns in the empty case because in the future we may have more complex nested columns and creating the entire tree with no data in each leaf seemed expensive. 

Also, there has been mention of supporting up to 2000 nested column entries. That is alot of immutable empty columns to build. That being said, there could be a case for compound columns that require some data to represent _emptiness_ and so I'm ok with not explicitly checking number of children if the `size()` is zero.

Perhaps `empty_like` can call `make_empty_strings_column` ?

Reply:
I think `empty_like` can be updated to _not_ create empty children. Just an empty parent.

I don't think that should be problematic, but we'll cross that bridge when we get there.

Reply:
See this commit for changes to empty_like: https://github.com/rapidsai/cudf/pull/3448/commits/0c537de304cbcaa9ad0d12755130f8109cc0ea00

Reply:
I'll revert that commit and use https://github.com/rapidsai/cudf/pull/3462 instead

Reply:
Done